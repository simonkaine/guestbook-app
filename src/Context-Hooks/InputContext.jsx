import { useContext, createContext, useState } from "react";

const inputContext = createContext();
const InputProvider = ({children}) => {

    const [input, setInput] = useState([]);

    return (
        <inputContext.Provider value={{input, setInput}}>
            {children}
        </inputContext.Provider>
    )
}

const useInputs = () => {
    const context = useContext(inputContext)
    if(context === undefined) {
        throw new Error("Yoooo, things ain't working with your inputs dawg..")
    } return context
}

export { InputProvider, useInputs }