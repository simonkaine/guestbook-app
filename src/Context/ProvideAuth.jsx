import dotenv from 'dotenv';
import { createContext, useState } from 'react';
dotenv.config();

export const authContext = createContext();
export function ProvideAuth({children}) {
    const [user, setUser] = useState(null);


    const login = (username, password) => {
        const loginIsSuccessful = username === process.env.REACT_APP_USERNAME 
        && 
        password === process.env.REACT_APP_PASSWORD;

        if(loginIsSuccessful) setUser({ username });
        return loginIsSuccessful;
    };

    const signout = (callback) => {
        setUser(null);
        callback();
    };

    return (
        <authContext.Provider value={{user, login, signout}}>
            {children}
        </authContext.Provider>
    )
}