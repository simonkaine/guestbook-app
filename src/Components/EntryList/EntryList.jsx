// import React from 'react';
import { useInputs } from '../../Context/InputContext';
import EntryInput from '../Entry/EntryInput';

export default function EntryList() {
    const { inputs } = useInputs();
    return (
        <div>
            <ul aria-label='list' style={{listStyle: 'none', border: '1px solid black'}}>
                {inputs.map((input) => {
                    return (
                        <div>
                        <li key={`${input.name} - ${input.message}`} style={{borderBottom: '.5px dotted red', marginBottom: '10px'}}>
                            <EntryInput entry={input} />
                        </li>
                        </div>
                    )
                })}
            </ul>
        </div>
    ) 
}
