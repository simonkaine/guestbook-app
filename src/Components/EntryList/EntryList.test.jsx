import React from 'react'
import { render, screen } from '@testing-library/react';
import EntryList from './EntryList';
import { InputProvider } from '../../Context/InputContext';

it('Should render a list of user entries', async() => {
    const {container} = render(
        <InputProvider>
            <EntryList />
        </InputProvider>
    )
    const list = await screen.findByRole(/list/i)
    expect(list).toBeInTheDocument(); 
    expect(container).toMatchSnapshot(); 
}) 