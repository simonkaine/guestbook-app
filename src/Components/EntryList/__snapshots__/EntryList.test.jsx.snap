// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a list of user entries 1`] = `
<div>
  <div>
    <ul
      aria-label="list"
      style="list-style: none; border: 1px solid black;"
    />
  </div>
</div>
`;
